### ISSUE TITLE
Failed to generate secret for proxy message inside wrong pod 

### ISSUE THREAD

**Issue by MaciekLeks** [istio/istio#33318]
**Bug description**
I've followed the example here [Configure mutual TLS origination for egress traffic](https://istio.io/latest/docs/tasks/traffic-management/egress/egress-gateway-tls-origination/#configure-mutual-tls-origination-for-egress-traffic) using sleep POD. But before that I've patched egress gateway according to instruction from here [Redeploy the egress gateway with the client certificates](https://istio.io/latest/docs/tasks/traffic-management/egress/egress-gateway-tls-origination/#redeploy-the-egress-gateway-with-the-client-certificates). Egress POD is configured properly. The example works OK. But, after having look at the sleep POD log I see repeating error not related to that istio-proxy:
```
2021-06-08T10:34:54.579510Z	info	ads	ADS: "@" sidecar~172.17.0.5~sleep-68c8b8969c-c6rf2.default~default.svc.cluster.local-444 terminated rpc error: code = Canceled desc = context canceled
2021-06-08T10:34:54.579829Z	warning	envoy config	StreamSecrets gRPC config stream closed: 2, failed to generate secret for file-root:/etc/istio/nginx-ca-certs/example.com.crt: open /etc/istio/nginx-ca-certs/example.com.crt: no such file or directory
2021-06-08T10:34:54.638238Z	error	cache	resource:file-cert:/etc/istio/nginx-client-certs/tls.crt~/etc/istio/nginx-client-certs/tls.key failed to generate secret for proxy from file: open /etc/istio/nginx-client-certs/tls.crt: no such file or directory
```

**Affected product area (please put an X in all that apply)**

[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[x] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Expected behavior**

Not seeing errors related to istio-proxy pod.

**Steps to reproduce the bug**

Follow the [example](https://istio.io/latest/docs/tasks/traffic-management/egress/egress-gateway-tls-origination/#configure-mutual-tls-origination-for-egress-traffic) instruction 

**Version**  
client version: 1.10.0
control plane version: 1.10.0
data plane version: 1.10.0 (3 proxies)

**How was Istio installed?**
By Operator

**Environment where the bug was observed (cloud vendor, OS, etc)**
minikube

[bug-report.tar.gz](https://github.com/istio/istio/files/6615806/bug-report.tar.gz)

**Comment by howardjohn** [comment]
This is an unfortunate limitation of the DestinationRule API; it applies fairly broadly. One thing you may wish to do is set `exportTo: ["."]` on the destination rule, which should limit its scope to istio-system.

BTW, I would recommend following https://istio.io/latest/docs/tasks/traffic-management/egress/egress-gateway-tls-origination-sds/ doc instead, we should probably remove the one you are looking at... you would have the same issue though

**Comment by MaciekLeks** [comment]
Thnak you. `exportTo: ["."]` will do its job.

**Comment by ewoks** [comment]
@howardjohn friendly ping if possible to clarify

**Comment by zirain** [comment]
https://istio.io/latest/docs/tasks/traffic-management/egress/egress-gateway-tls-origination/

**Comment by ewoks** [comment]
@howardjohn could you please confirm one or the other approach?

**Comment by howardjohn** [comment]
I believe https://istio.io/latest/docs/tasks/traffic-management/egress/egress-gateway-tls-origination-sds was renamed to https://istio.io/latest/docs/tasks/traffic-management/egress/egress-gateway-tls-origination/, and the original https://istio.io/latest/docs/tasks/traffic-management/egress/egress-gateway-tls-origination/ was removed.

**Comment by ewoks** [comment]
@howardjohn  just that the content of 1st and 2nd link you posted is not the same, and 2nd link doesn't even mention "SDS" nor "discovery" :(

**Comment by howardjohn** [comment]
Yes, those are internal terms that were removed from the documentation since they are not relevant to users.
